```
TASK [ecs_cluster : create ECS service definition with network configuration] ***
task path: /root/ansible/test/integration/targets/ecs_cluster/playbooks/roles/ecs_cluster/tasks/main.yml:417
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c 'echo ~root && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390 `" && echo ansible-tmp-1542968685.9855068-144333392449390="` echo /root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390 `" ) && sleep 0'
Using module file /root/ansible/lib/ansible/modules/cloud/amazon/ecs_service.py
<127.0.0.1> PUT /root/.ansible/tmp/ansible-local-883hl1l4e9r/tmp6nkzmr_9 TO /root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/AnsiballZ_ecs_service.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/ /root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/AnsiballZ_ecs_service.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/tmp/ansible-test-coverage-r9e6n06y/coverage/injector.py /root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/AnsiballZ_ecs_service.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/AnsiballZ_ecs_service.py", line 113, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/AnsiballZ_ecs_service.py", line 105, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/AnsiballZ_ecs_service.py", line 48, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File "/usr/lib/python3.6/imp.py", line 235, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.6/imp.py", line 170, in load_source
    module = _exec(spec, sys.modules[name])
  File "<frozen importlib._bootstrap>", line 618, in _exec
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmp/ansible_ecs_service_payload_1qonwwiu/__main__.py", line 621, in <module>
  File "/tmp/ansible_ecs_service_payload_1qonwwiu/__main__.py", line 566, in main
  File "/tmp/ansible_ecs_service_payload_1qonwwiu/__main__.py", line 411, in create_service
  File "/usr/lib/python3.6/site-packages/botocore/client.py", line 320, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/lib/python3.6/site-packages/botocore/client.py", line 597, in _make_api_call
    api_params, operation_model, context=request_context)
  File "/usr/lib/python3.6/site-packages/botocore/client.py", line 633, in _convert_to_request_dict
    api_params, operation_model)
  File "/usr/lib/python3.6/site-packages/botocore/validate.py", line 291, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter healthCheckGracePeriodSeconds, value: None, type: <class 'NoneType'>, valid types: <class 'int'>

fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/AnsiballZ_ecs_service.py\", line 113, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/AnsiballZ_ecs_service.py\", line 105, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1542968685.9855068-144333392449390/AnsiballZ_ecs_service.py\", line 48, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/usr/lib/python3.6/imp.py\", line 235, in load_module\n    return load_source(name, filename, file)\n  File \"/usr/lib/python3.6/imp.py\", line 170, in load_source\n    module = _exec(spec, sys.modules[name])\n  File \"<frozen importlib._bootstrap>\", line 618, in _exec\n  File \"<frozen importlib._bootstrap_external>\", line 678, in exec_module\n  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n  File \"/tmp/ansible_ecs_service_payload_1qonwwiu/__main__.py\", line 621, in <module>\n  File \"/tmp/ansible_ecs_service_payload_1qonwwiu/__main__.py\", line 566, in main\n  File \"/tmp/ansible_ecs_service_payload_1qonwwiu/__main__.py\", line 411, in create_service\n  File \"/usr/lib/python3.6/site-packages/botocore/client.py\", line 320, in _api_call\n    return self._make_api_call(operation_name, kwargs)\n  File \"/usr/lib/python3.6/site-packages/botocore/client.py\", line 597, in _make_api_call\n    api_params, operation_model, context=request_context)\n  File \"/usr/lib/python3.6/site-packages/botocore/client.py\", line 633, in _convert_to_request_dict\n    api_params, operation_model)\n  File \"/usr/lib/python3.6/site-packages/botocore/validate.py\", line 291, in serialize_to_request\n    raise ParamValidationError(report=report.generate_report())\nbotocore.exceptions.ParamValidationError: Parameter validation failed:\nInvalid type for parameter healthCheckGracePeriodSeconds, value: None, type: <class 'NoneType'>, valid types: <class 'int'>\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

--- a/lib/ansible/modules/cloud/amazon/ecs_service.py
+++ b/lib/ansible/modules/cloud/amazon/ecs_service.py
@@ -406,7 +406,7 @@ class EcsServiceManager:
             params['networkConfiguration'] = network_configuration
         if launch_type:
             params['launchType'] = launch_type
-        if self.health_check_setable(params):
+        if self.health_check_setable(params) and health_check_grace_period_seconds is not None:
             params['healthCheckGracePeriodSeconds'] = health_check_grace_period_seconds
         response = self.ecs.create_service(**params)
         return self.jsonize(response['service'])

